config/initializers/omniauth.rb

```diff
@@ -36,10 +35,23 @@
   Rails.application.config.middleware.use OmniAuth::Strategies::OpenIDConnect, options
 end
 
-# Identity setup
 Rails.application.config.middleware.use OmniAuth::Builder do
-  provider :identity,
-           ENV.fetch("IDENTITY_CLIENT_ID"),
-           ENV.fetch("IDENTITY_CLIENT_SECRET"),
-           { path_prefix: "/qualifications/users/auth" }
+  provider :openid_connect,
+           name: :identity,
+           callback_path: "/qualifications/users/auth/identity/callback",
+           client_options: {
+             host: URI(ENV.fetch("IDENTITY_API_DOMAIN")).host,
+             identifier: ENV.fetch("IDENTITY_CLIENT_ID"),
+             port: 443,
+             redirect_uri: ENV.fetch("IDENTITY_REDIRECT_URI"),
+             scheme: "https",
+             secret: ENV.fetch("IDENTITY_CLIENT_SECRET")
+           },
+           discovery: true,
+           issuer: ENV.fetch("IDENTITY_API_DOMAIN"),
+           path_prefix: "/qualifications/users/auth",
+           pkce: true,
+           post_logout_redirect_uri: "#{ENV.fetch("HOSTING_DOMAIN")}/qualifications/sign-out",
+           response_type: :code,
+           scope: %w[email openid profile dqt:read]
```

Comment:
Does making this switch allow us to simplify the existing logout implementation at all?